daemon/qrexec-client.c

```diff
@@ -280,16 +288,21 @@ static void send_exit_code(libvchan_t *vchan, int status)
     }
 }
 
-static void handle_input(libvchan_t *vchan)
+static void handle_input(libvchan_t *vchan, int data_protocol_version)
 {
-    char buf[MAX_DATA_CHUNK];
+    char buf[MAX_DATA_CHUNK_V3];
     int ret;
     size_t max_len;
     struct msg_header hdr;
 
-    max_len = libvchan_buffer_space(vchan)-sizeof(hdr);
+    max_len = libvchan_buffer_space(vchan);
+    if (max_len < sizeof(hdr))
+        return;
```

Comment:
Don't we want some hard exit in this strange case?

Reply:
This isn't strange case. It may mean remote side haven't retrieved previous data yet.

Reply:
Oh, got confused about the meaning of `libvchan_buffer_space`.